sktime/transformations/series/difference.py

```diff
@@ -145,7 +145,7 @@ def _inverse_diff(X, lags, X_diff_seq=None):
         X_diff_orig = X_diff_seq[len(lags)]
         X_ix_shift = X.index.shift(-lag_last)
         X_update = X_diff_orig.loc[X_ix_shift.intersection(X_diff_orig.index)]
-
+        X.loc[X.index.intersection(X_diff_orig.index[:lag_last])] = np.nan
```

Comment:
instead of taking "first lag_last", should we work with `_shift` instead?

Because the index might not be contiguous. That is, instead of the intersection, take the difference with `_shift` of `lag_last`?

Also see https://github.com/sktime/sktime/pull/4984

Reply:
Good point. I need to test it with the shift. Probably the best is if I rebase on the branch of the linked PR.

Thank you for linking the other PR.

Reply:
Ok, I rebased. 
Happy to add a comment if this line is too complicated.

Reply:
yes, I think comments in the code would be helpful!
I had myself some problems reading this again after a year.

Reply:
addressed